File: BraintreeCore/src/main/java/com/braintreepayments/api/UnionPayConfiguration.java
package com.braintreepayments.api;


import org.json.JSONObject;

class UnionPayConfiguration {

    private static final String ENABLED = "enabled";

    private boolean enabled;

    static UnionPayConfiguration fromJson(JSONObject json) {
        if (json == null) {
            json = new JSONObject();
        }

        UnionPayConfiguration unionPayConfiguration = new UnionPayConfiguration();
        unionPayConfiguration.enabled = json.optBoolean(ENABLED, false);

        return unionPayConfiguration;
    }

    /**
     * Determines if UnionPay is available to be used
     *
     * @return boolean if UnionPay is enabled, and available to be used
     */
    boolean isEnabled() {
        return enabled;
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/UnionPayConfiguration.kt
package com.braintreepayments.api

import org.json.JSONObject

/**
 * Determines if UnionPay is available to be used
 * @property isEnabled boolean if UnionPay is enabled, and available to be used
 */
internal data class UnionPayConfiguration(val isEnabled: Boolean) {

    constructor(json: JSONObject?) : this(json?.optBoolean(ENABLED, false) ?: false)

    companion object {
        private const val ENABLED = "enabled"
    }
}
